### ISSUE TITLE
Game Capture gets progressively delayed based on DXCache size (Monster Hunter World/Nvidia)

### ISSUE THREAD

**Issue by Miltrivd** [obsproject/obs-studio#7208]
### Operating System Info

Windows 10

### OBS Studio Version

28.0.0-rc2

### OBS Studio Log URL

https://obsproject.com/logs/IPSU-Wk2RuOlAjxj

### Expected Behavior

Game Capture hooks as soon as the game graphics start.

### Current Behavior

Game Capture starts getting delayed by a couple seconds progressively taking longer to capture, can go into 15 minutes. At some point it also "freezes" the game (audio continues normally but gets stuck on a black screen unable to progress to the main menu until the Game Capture hooks successfully).

### Steps to Reproduce

1. Have shader cache enabled on Nvidia control panel (on by default)
2. Seems to show up when the shader cache growths near the 1 GB mark.
3. Have the scene with Game Capture, active and launch the game (Monster Hunter World). 

### Anything else we should know?

- I've only seen this behavior exclusively on Monster Hunter World with both the DX12 and DX11 renderers. Other DX9, DX11 and DX12 games do not show this issue so far)
- This did not happen with AMD drivers (last year I was on a 390 with shader cache forced on via Windows registry).
- [I reported this a few weeks ago on the OBS forums with the latest stable release](https://obsproject.com/forum/threads/obs-takes-up-to-3-mins-for-capture-to-start-freezes-game-in-the-meantime.158225/#post-580216) and was on a machine with no software or hardware modifications when the issue suddenly popped up after using it for months with no odd behavior.

Since then I did a clean Windows 10 x64 21H2 installation and used clean install 516.94 Nvidia drivers but before wiping my previous Windows installation I realized cleaning the cache immediately reverted OBS to normal behavior. 

Now, right after I uploaded the previous log, I manually deleted the files on \AppData\Local\NVIDIA\DXCache\ and the issue immediately went away:

https://obsproject.com/logs/aTi_mpV-seK2L4hr

I don't know if this a specific Nvidia + OBS issue or a specific Nvidia + OBS + MHW issue. I have the crude workaround of deleting cache to fix it but felt it was notable to report in case it shows up on other games down the line since with DX12/Vulkan games in particular rebuilding cache can lead to stuttering while playing.

**Comment by Fenrirthviti** [comment]
I will try and make time to test this, but since there has been quite a few NVIDIA driver updates since then, can you retest on your end as well with the latest drivers? The 516.x drivers were fairly notorious for a multitude of issues, so I wouldn't be surprised if this is already fixed on NVIDIA's side.

**Comment by Miltrivd** [comment]
One thing I forgot to mention, since I found out at a later date; this only happened when "Fullscreen" mode was selected, it did not seem to trigger on "Borderless Window", which is how I "fixed" the issue.

Tested it just now on 522.25, and OBS 29.0.0 64-bits. My shader cache is fully built for the game. Both DX11 and DX12 modes do not show the issue anymore.

So this doesn't seem to be an issue any longer.
